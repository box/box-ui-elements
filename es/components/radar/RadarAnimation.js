const _excluded = ["children", "className", "targetWrapperClassName", "constrainToScrollParent", "constrainToWindow", "position", "isShown", "offset", "tetherElementClassName"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import * as React from 'react';
import classNames from 'classnames';
import uniqueId from 'lodash/uniqueId';
import TetherComponent from 'react-tether';
import TetherPosition from '../../common/tether-positions';
import './RadarAnimation.scss';
export let RadarAnimationPosition = /*#__PURE__*/function (RadarAnimationPosition) {
  RadarAnimationPosition["BOTTOM_CENTER"] = "bottom-center";
  RadarAnimationPosition["BOTTOM_LEFT"] = "bottom-left";
  RadarAnimationPosition["BOTTOM_RIGHT"] = "bottom-right";
  RadarAnimationPosition["MIDDLE_CENTER"] = "middle-center";
  RadarAnimationPosition["MIDDLE_LEFT"] = "middle-left";
  RadarAnimationPosition["MIDDLE_RIGHT"] = "middle-right";
  RadarAnimationPosition["TOP_CENTER"] = "top-center";
  RadarAnimationPosition["TOP_LEFT"] = "top-left";
  RadarAnimationPosition["TOP_RIGHT"] = "top-right";
  return RadarAnimationPosition;
}({});
const positions = {
  [RadarAnimationPosition.BOTTOM_CENTER]: {
    attachment: TetherPosition.TOP_CENTER,
    targetAttachment: TetherPosition.BOTTOM_CENTER
  },
  [RadarAnimationPosition.BOTTOM_LEFT]: {
    attachment: TetherPosition.TOP_LEFT,
    targetAttachment: TetherPosition.BOTTOM_LEFT
  },
  [RadarAnimationPosition.BOTTOM_RIGHT]: {
    attachment: TetherPosition.TOP_RIGHT,
    targetAttachment: TetherPosition.BOTTOM_RIGHT
  },
  [RadarAnimationPosition.MIDDLE_CENTER]: {
    attachment: TetherPosition.MIDDLE_CENTER,
    targetAttachment: TetherPosition.MIDDLE_CENTER
  },
  [RadarAnimationPosition.MIDDLE_LEFT]: {
    attachment: TetherPosition.MIDDLE_RIGHT,
    targetAttachment: TetherPosition.MIDDLE_LEFT
  },
  [RadarAnimationPosition.MIDDLE_RIGHT]: {
    attachment: TetherPosition.MIDDLE_LEFT,
    targetAttachment: TetherPosition.MIDDLE_RIGHT
  },
  [RadarAnimationPosition.TOP_CENTER]: {
    attachment: TetherPosition.BOTTOM_CENTER,
    targetAttachment: TetherPosition.TOP_CENTER
  },
  [RadarAnimationPosition.TOP_LEFT]: {
    attachment: TetherPosition.BOTTOM_LEFT,
    targetAttachment: TetherPosition.TOP_LEFT
  },
  [RadarAnimationPosition.TOP_RIGHT]: {
    attachment: TetherPosition.BOTTOM_RIGHT,
    targetAttachment: TetherPosition.TOP_RIGHT
  }
};
class RadarAnimation extends React.Component {
  constructor(...args) {
    super(...args);
    _defineProperty(this, "tetherRef", /*#__PURE__*/React.createRef());
    _defineProperty(this, "radarAnimationID", uniqueId('radarAnimation'));
    // Instance API: Forces the radar to be repositioned
    _defineProperty(this, "position", () => {
      const {
        isShown
      } = this.props;
      if (this.tetherRef.current && isShown) {
        this.tetherRef.current.position();
      }
    });
  }
  render() {
    const _this$props = this.props,
      {
        children,
        className = '',
        targetWrapperClassName,
        constrainToScrollParent,
        constrainToWindow,
        position,
        isShown,
        offset,
        tetherElementClassName
      } = _this$props,
      rest = _objectWithoutProperties(_this$props, _excluded);
    const constraints = [];
    if (constrainToScrollParent) {
      constraints.push({
        to: 'scrollParent',
        attachment: 'together'
      });
    }
    if (constrainToWindow) {
      constraints.push({
        to: 'window',
        attachment: 'together'
      });
    }
    const {
      attachment,
      targetAttachment
    } = positions[position];
    const child = React.Children.only(children);
    const referenceElement = /*#__PURE__*/React.cloneElement(child, {
      'aria-describedby': this.radarAnimationID
    });
    const tetherProps = {
      attachment,
      classPrefix: 'radar-animation',
      constraints,
      enabled: isShown,
      targetAttachment
    };
    if (tetherElementClassName) {
      tetherProps.className = tetherElementClassName;
    }
    if (offset) {
      tetherProps.offset = offset;
    }
    return /*#__PURE__*/React.createElement(TetherComponent, _extends({
      ref: this.tetherRef
    }, tetherProps, {
      renderTarget: ref => /*#__PURE__*/React.createElement("div", {
        ref: ref,
        className: classNames('bdl-RadarAnimation-target', targetWrapperClassName)
      }, referenceElement),
      renderElement: ref => /*#__PURE__*/React.createElement("div", _extends({
        ref: ref,
        className: `radar ${className}`,
        id: this.radarAnimationID
      }, rest), /*#__PURE__*/React.createElement("div", {
        className: "radar-dot"
      }), /*#__PURE__*/React.createElement("div", {
        className: "radar-circle"
      }))
    }));
  }
}
_defineProperty(RadarAnimation, "defaultProps", {
  constrainToScrollParent: false,
  constrainToWindow: true,
  isShown: true,
  position: RadarAnimationPosition.MIDDLE_RIGHT
});
export default RadarAnimation;
//# sourceMappingURL=RadarAnimation.js.map