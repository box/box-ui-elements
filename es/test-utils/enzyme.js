function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map