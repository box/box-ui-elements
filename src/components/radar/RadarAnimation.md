**bottom-center**

```js
<RadarAnimation position="bottom-center">
  <Button>Bottom Center</Button>
</RadarAnimation>
```

**bottom-left**

```js
<RadarAnimation position="bottom-left">
  <Button>Bottom Left</Button>
</RadarAnimation>
```

**bottom-right**

```js
<RadarAnimation position="bottom-right">
  <Button>Bottom Right</Button>
</RadarAnimation>
```

**middle-center**

```js
<RadarAnimation position="middle-center">
  <Button>Middle Center</Button>
</RadarAnimation>
```

**middle-left**

```js
<RadarAnimation position="middle-left">
  <Button>Middle Left</Button>
</RadarAnimation>
```

**middle-right**

```js
<RadarAnimation position="middle-right">
  <Button>Middle Right</Button>
</RadarAnimation>
```

**top-center**

```js
<RadarAnimation position="top-center">
  <Button>Top Center</Button>
</RadarAnimation>
```

**top-left**

```js
<RadarAnimation position="top-left">
  <Button>Top Left</Button>
</RadarAnimation>
```

**top-right**

```js
<RadarAnimation position="top-right">
  <Button>Top Right</Button>
</RadarAnimation>
```
