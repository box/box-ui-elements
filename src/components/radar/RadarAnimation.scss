@import '../../styles/variables';
$radar-animation-offset: 1px;

.radar {
    position: relative;

    .radar-dot::before {
        position: absolute;
        top: -4px;
        left: -4px;
        z-index: $overlay-z-index + 3;
        width: 8px;
        height: 8px;
        background-color: $bdl-box-blue;
        border: 2px solid $white;
        border-radius: 50%;
        content: ' ';
    }

    .radar-dot::after {
        position: absolute;
        z-index: $overlay-z-index + 2;
        width: 10px;
        height: 10px;
        background-color: $bdl-box-blue;
        border-radius: 50%;
        box-shadow: 0 0 10px rgba(0, 0, 0, .3) inset;
        animation-name: ripple;
        animation-duration: 1s;
        animation-timing-function: ease;
        animation-delay: 0s;
        animation-iteration-count: infinite;
        content: ' ';
    }

    .radar-circle::before {
        position: absolute;
        top: -20px;
        left: -20px;
        z-index: $overlay-z-index + 2;
        width: 40px;
        height: 40px;
        border: 1px solid $bdl-box-blue;
        border-radius: 50%;
        content: ' ';
    }

    .radar-circle::after {
        position: absolute;
        top: -20px;
        left: -20px;
        z-index: $overlay-z-index + 2;
        width: 40px;
        height: 40px;
        background-color: $bdl-box-blue;
        border-radius: 50%;
        opacity: .3;
        content: ' ';
    }

    @keyframes ripple {
        0% {
            top: 0;
            left: 0;
            width: 0;
            height: 0;
            opacity: .75;
        }

        100% {
            top: -20px;
            left: -20px;
            width: 40px;
            height: 40px;
            opacity: 0;
        }
    }

    .radar-animation-target-attached-bottom.radar-animation-target-attached-center > & {
        margin-top: $radar-animation-offset;
    }

    .radar-animation-target-attached-bottom.radar-animation-target-attached-left > & {
        margin-top: $radar-animation-offset;
    }

    .radar-animation-target-attached-bottom.radar-animation-target-attached-right > & {
        margin-top: $radar-animation-offset;
    }

    .radar-animation-target-attached-middle.radar-animation-target-attached-right > & {
        margin-left: $radar-animation-offset;
    }

    .radar-animation-target-attached-middle.radar-animation-target-attached-center > & {
        margin-top: $radar-animation-offset;
    }

    .radar-animation-target-attached-middle.radar-animation-target-attached-left > & {
        margin-right: $radar-animation-offset;
    }

    .radar-animation-target-attached-top.radar-animation-target-attached-center > & {
        margin-bottom: $radar-animation-offset;
    }

    .radar-animation-target-attached-top.radar-animation-target-attached-left > & {
        margin-bottom: $radar-animation-offset;
    }

    .radar-animation-target-attached-top.radar-animation-target-attached-right > & {
        margin-bottom: $radar-animation-offset;
    }
}

.radar-animation-element {
    z-index: $overlay-z-index;
}
