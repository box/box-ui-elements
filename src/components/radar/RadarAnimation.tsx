import * as React from 'react';
import uniqueId from 'lodash/uniqueId';
import TetherComponent from 'react-tether';

import './RadarAnimation.scss';

export enum RadarAnimationPosition {
    BOTTOM_CENTER = 'bottom-center',
    BOTTOM_LEFT = 'bottom-left',
    BOTTOM_RIGHT = 'bottom-right',
    MIDDLE_CENTER = 'middle-center',
    MIDDLE_LEFT = 'middle-left',
    MIDDLE_RIGHT = 'middle-right',
    TOP_CENTER = 'top-center',
    TOP_LEFT = 'top-left',
    TOP_RIGHT = 'top-right',
}

const positions = {
    [RadarAnimationPosition.BOTTOM_CENTER]: {
        attachment: 'top center',
        targetAttachment: 'bottom center',
    },
    [RadarAnimationPosition.BOTTOM_LEFT]: {
        attachment: 'top left',
        targetAttachment: 'bottom left',
    },
    [RadarAnimationPosition.BOTTOM_RIGHT]: {
        attachment: 'top right',
        targetAttachment: 'bottom right',
    },
    [RadarAnimationPosition.MIDDLE_CENTER]: {
        attachment: 'middle center',
        targetAttachment: 'middle center',
    },
    [RadarAnimationPosition.MIDDLE_LEFT]: {
        attachment: 'middle right',
        targetAttachment: 'middle left',
    },
    [RadarAnimationPosition.MIDDLE_RIGHT]: {
        attachment: 'middle left',
        targetAttachment: 'middle right',
    },
    [RadarAnimationPosition.TOP_CENTER]: {
        attachment: 'bottom center',
        targetAttachment: 'top center',
    },
    [RadarAnimationPosition.TOP_LEFT]: {
        attachment: 'bottom left',
        targetAttachment: 'top left',
    },
    [RadarAnimationPosition.TOP_RIGHT]: {
        attachment: 'bottom right',
        targetAttachment: 'top right',
    },
};

export interface RadarAnimationProps {
    /** A React element to put the radar on */
    children: React.ReactChild;
    /** A CSS class for the radar */
    className?: string;
    /** Whether to constrain the radar to the element's scroll parent. Defaults to `false` */
    constrainToScrollParent: boolean;
    /** Whether to constrain the radar to window. Defaults to `true` */
    constrainToWindow: boolean;
    /** Forces the radar to be shown or hidden - defaults to true */
    isShown: boolean;
    /** A string of the form 'vert-offset horiz-offset' which controls positioning */
    offset?: string;
    /** Where to position the radar relative to the wrapped component */
    position: RadarAnimationPosition;
}

class RadarAnimation extends React.Component<RadarAnimationProps> {
    tetherRef = React.createRef<TetherComponent>();

    radarAnimationID = uniqueId('radarAnimation');

    static defaultProps = {
        constrainToScrollParent: false,
        constrainToWindow: true,
        isShown: true,
        position: RadarAnimationPosition.MIDDLE_RIGHT,
    };

    // Instance API: Forces the radar to be repositioned
    position = () => {
        const { isShown } = this.props;
        if (this.tetherRef.current && isShown) {
            this.tetherRef.current.position();
        }
    };

    render() {
        const {
            children,
            className = '',
            constrainToScrollParent,
            constrainToWindow,
            position,
            isShown,
            offset,
            ...rest
        } = this.props;

        const constraints = [];
        if (constrainToScrollParent) {
            constraints.push({
                to: 'scrollParent',
                attachment: 'together',
            });
        }
        if (constrainToWindow) {
            constraints.push({
                to: 'window',
                attachment: 'together',
            });
        }

        const { attachment, targetAttachment } = positions[position];
        const child = React.Children.only(children);
        const referenceElement = React.cloneElement(child as React.ReactElement, {
            'aria-describedby': this.radarAnimationID,
        });

        // Typescript defs seem busted for older versions of react-tether
        const tetherProps: {
            attachment: string;
            classPrefix: string;
            constraints: {};
            targetAttachment: string;
            offset?: string;
        } = {
            attachment,
            classPrefix: 'radar-animation',
            constraints,
            targetAttachment,
        };

        if (offset) {
            tetherProps.offset = offset;
        }

        return (
            <TetherComponent ref={this.tetherRef} {...tetherProps}>
                {referenceElement}
                {isShown && (
                    <div className={`radar ${className}`} id={this.radarAnimationID} {...rest}>
                        <div className="radar-dot" />
                        <div className="radar-circle" />
                    </div>
                )}
            </TetherComponent>
        );
    }
}

export default RadarAnimation;
