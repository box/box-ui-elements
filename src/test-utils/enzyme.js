/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
