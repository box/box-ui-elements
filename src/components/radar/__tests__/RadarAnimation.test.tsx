import React from 'react';
import { shallow } from 'enzyme';
import RadarAnimation, { RadarAnimationProps, RadarAnimationPosition } from '../RadarAnimation';

describe('components/radar/RadarAnimation', () => {
    const getWrapper = (props: {}) =>
        shallow<RadarAnimation>(
            <RadarAnimation {...props}>
                <div>Hello</div>
            </RadarAnimation>,
        );
    [
        {
            // description:
            //     'should render correctly with bottom-center positioning',
            position: RadarAnimationPosition.BOTTOM_CENTER,
        },
        {
            // description: 'should render correctly with bottom-left positioning',
            position: RadarAnimationPosition.BOTTOM_LEFT,
        },
        {
            // description:
            //     'should render correctly with bottom-right positioning',
            position: RadarAnimationPosition.BOTTOM_RIGHT,
        },
        {
            // description:
            //     'should render correctly with middle-center positioning',
            position: RadarAnimationPosition.MIDDLE_CENTER,
        },
        {
            // description: 'should render correctly with middle-left positioning',
            position: RadarAnimationPosition.MIDDLE_LEFT,
        },
        {
            // description:
            //     'should render correctly with middle-right positioning',
            position: RadarAnimationPosition.MIDDLE_RIGHT,
        },
        {
            // description: 'should render correctly with top-center positioning',
            position: RadarAnimationPosition.TOP_CENTER,
        },
        {
            // description: 'should render correctly with top-left positioning',
            position: RadarAnimationPosition.TOP_LEFT,
        },
        {
            // description: 'should render correctly with top-right positioning',
            position: RadarAnimationPosition.TOP_RIGHT,
        },
    ].forEach(({ position }) => {
        test(`should render correctly with ${position} positioning`, () => {
            const wrapper = getWrapper({ position } as RadarAnimationProps);
            expect(wrapper).toMatchSnapshot();
        });
    });

    test('should spread the rest of the props to the radar node', () => {
        const wrapper = getWrapper({
            'data-resin-target': 'radaranimation1',
        });
        expect(wrapper.find('.radar').prop('data-resin-target')).toBe('radaranimation1');
    });

    test('should render with custom offset when provided', () => {
        const offset = '0 10px';
        const wrapper = getWrapper({
            offset,
        });

        expect(wrapper.prop('offset')).toEqual(offset);
    });

    describe('isShown', () => {
        test('should be shown when isShown is not provided', () => {
            expect(
                getWrapper({} as RadarAnimationProps)
                    .find('.radar')
                    .exists(),
            ).toBe(true);
        });

        test('should be shown when isShown is true', () => {
            expect(
                getWrapper({ isShown: true } as RadarAnimationProps)
                    .find('.radar')
                    .exists(),
            ).toBe(true);
        });

        test('should not be shown when isShown is false', () => {
            expect(
                getWrapper({ isShown: false } as RadarAnimationProps)
                    .find('.radar')
                    .exists(),
            ).toBe(false);
        });
    });

    describe('position instance method', () => {
        test.each([true, false])('should only position the tether when shown', isShown => {
            const positionTetherMock = jest.fn();

            const wrapper = getWrapper({ isShown } as RadarAnimationProps);
            // @ts-ignore: react-tether shenanigans
            wrapper.instance().tetherRef = { current: { position: positionTetherMock } };

            wrapper.instance().position();

            expect(positionTetherMock).toHaveBeenCalledTimes(isShown ? 1 : 0);
        });
    });
});
