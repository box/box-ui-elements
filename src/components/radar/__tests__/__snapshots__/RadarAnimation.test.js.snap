// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/radar/RadarAnimation should render correctly with bottom-center positioning 1`] = `
<TetherComponent
  attachment="top center"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="bottom center"
>
  <div
    aria-describedby="radarAnimation1"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation1"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;

exports[`components/radar/RadarAnimation should render correctly with bottom-left positioning 1`] = `
<TetherComponent
  attachment="top left"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="bottom left"
>
  <div
    aria-describedby="radarAnimation2"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation2"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;

exports[`components/radar/RadarAnimation should render correctly with bottom-right positioning 1`] = `
<TetherComponent
  attachment="top right"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="bottom right"
>
  <div
    aria-describedby="radarAnimation3"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation3"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;

exports[`components/radar/RadarAnimation should render correctly with middle-center positioning 1`] = `
<TetherComponent
  attachment="middle center"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="middle center"
>
  <div
    aria-describedby="radarAnimation4"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation4"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;

exports[`components/radar/RadarAnimation should render correctly with middle-left positioning 1`] = `
<TetherComponent
  attachment="middle right"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="middle left"
>
  <div
    aria-describedby="radarAnimation5"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation5"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;

exports[`components/radar/RadarAnimation should render correctly with middle-right positioning 1`] = `
<TetherComponent
  attachment="middle left"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="middle right"
>
  <div
    aria-describedby="radarAnimation6"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation6"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;

exports[`components/radar/RadarAnimation should render correctly with top-center positioning 1`] = `
<TetherComponent
  attachment="bottom center"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="top center"
>
  <div
    aria-describedby="radarAnimation7"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation7"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;

exports[`components/radar/RadarAnimation should render correctly with top-left positioning 1`] = `
<TetherComponent
  attachment="bottom left"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="top left"
>
  <div
    aria-describedby="radarAnimation8"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation8"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;

exports[`components/radar/RadarAnimation should render correctly with top-right positioning 1`] = `
<TetherComponent
  attachment="bottom right"
  classPrefix="radar-animation"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="top right"
>
  <div
    aria-describedby="radarAnimation9"
  >
    Hello
  </div>
  <div
    className="radar "
    id="radarAnimation9"
  >
    <div
      className="radar-dot"
    />
    <div
      className="radar-circle"
    />
  </div>
</TetherComponent>
`;
